spec/requests/v1/reviews_spec.rb

```diff
@@ -39,62 +44,108 @@
 
   describe "/reviews" do
     context 'with proper privileges' do
+      let(:user) { approving_user }
+
+      it "adds comment" do
+        Apartment::Tenant.switch(account.tenant) do
+          expect do
+            post "/api/v1/tenant/#{account.tenant}/work/#{work.id}/reviews", params: {
+              name: "comment_only",
+              comment: "can manage workflow from api"
+            }, headers: { "Cookie" => response['Set-Cookie'] }
+          end.to change { work.reload.to_sipity_entity.comments.count }.by(1)
+          expect(work.reload.to_sipity_entity.comments.count).to eq 1
+          expect(response.status).to eq(200)
+          json_response = JSON.parse(response.body)
+          expect(json_response['comments'][0]['comment']).to be_present
+          expect(json_response['comments'][0]['updated_at']).to be_present
+          expect(json_response['workflow_status']).to eq 'pending_review'
+        end
+      end
+
+      it 'changes state' do
+        Apartment::Tenant.switch(account.tenant) do
+          expect do
+            post "/api/v1/tenant/#{account.tenant}/work/#{work.id}/reviews", params: {
+              name: "approve",
+              comment: "can manage workflow from api"
+            }, headers: { "Cookie" => response['Set-Cookie'] }
+          end.to change { work.to_sipity_entity.reload.workflow_state_name }.from("pending_review").to("deposited")
+
+          expect(response.status).to eq(200)
+          json_response = JSON.parse(response.body)
+          expect(json_response['comments'][0]['comment']).to be_present
+          expect(json_response['comments'][0]['updated_at']).to be_present
+          expect(json_response['workflow_status']).to eq 'deposited'
+        end
+      end
+    end
+
+    context 'without proper privileges' do
+      let(:user) { create(:user) }
+
       before do
-        post hyku_api.v1_tenant_users_login_path(tenant_id: account.tenant), params: {
-          email: approving_user.email,
-          password: approving_user.password,
-          expire: 2
-        }
+        Apartment::Tenant.switch(account.tenant) { user }
```

Comment:
I don't know if these before blocks are necessary anymore because the user should have been created in the tenant in the login post request in the parent before block.

Reply:
You are right! They seem to pass without the switch calls. Pushing a new version now, thanks!